packages/components/src/Socket.js

```diff
@@ -0,0 +1,96 @@
+import { useEffect } from 'react'
+
+const STORAGE_INDEX = 'mdx-slide'
+const STORAGE_STEP = 'mdx-step'
+function createWS() {
+  if (typeof window === 'object') {
+    const url = window.location.origin
+      .replace('https:', 'wss:')
+      .replace('http:', 'ws:')
+    const socket = new WebSocket(`${url}/__socket`)
+    socket.addEventListener('open', function() {
+      console.log('ws:connected !!')
+    })
+    socket.addEventListener('error', function(event) {
+      console.log('ws:error !!', event)
+    })
+    return socket
+  }
+}
+const socket = createWS()
+function sendMessage(message) {
+  if (socket.readyState === WebSocket.OPEN) {
+    socket.send(JSON.stringify(message))
+  } else {
+    console.warn(`socket is not ready, status :${socket.readyState}`)
+  }
+}
+function useSocketMessage({ goto, update, index, step }) {
+  useEffect(
+    function() {
+      function handler(event) {
+        const data = JSON.parse(event.data)
+        console.log('Message from server ', data)
+        if (data.type === 'broadcast') {
+          const value = data.payload.value - 0
+          if (data.payload.type === STORAGE_INDEX && value !== index) {
+            localStorage.setItem(STORAGE_INDEX, value)
```

Comment:
I might be misunderstanding how this works with the existing code, but doesn't the `Storage` component handle this already?

Reply:
Yes, `Storage` does the same job. Sometimes(maybe by changing slides at a fast speed) `localstorage` goes out of sync, so first updating the storage & emitting the changes to react.
I couldn't reproduce the exact steps. so commenting these lines as of now.